jwala-webservices/src/main/java/com/cerner/jwala/ws/rest/v1/service/resource/impl/ResourceServiceRestImpl.java

```diff
@@ -132,17 +132,29 @@ public Response createResource(final String deployFilename, final CreateResource
         // TODO pass down single param from UI to designate external properties
         final boolean isExternalProperty = createResourceParam.getGroup() == null && createResourceParam.getJvm() == null && createResourceParam.getWebApp() == null && createResourceParam.getWebServer() == null;
 
-        if (attachments == null || !isExternalProperty && attachments.size() != CREATE_RESOURCE_ATTACHMENT_SIZE) {
-            return ResponseBuilder.notOk(Response.Status.INTERNAL_SERVER_ERROR, new FaultCodeException(
-                    FaultType.INVALID_NUMBER_OF_ATTACHMENTS,
-                    "Invalid number of attachments! " + CREATE_RESOURCE_ATTACHMENT_SIZE + " attachments is expected by the service."));
+        if (attachments == null || !isExternalProperty && attachments.size() == 0 ) {
+            LOGGER.error("Expected non-empty attachments. Returning without creating Resource.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+
+        if (attachments != null){
```

Comment:
Should the method exit early if there are no attachments provided? Would it make more sense to throw an exception here if there are no attachments?